pkg/services/loginattempt/login_attempt.go

```diff
@@ -5,11 +5,14 @@ import (
 )
 
 type Service interface {
-	// Add adds a new login attempt record for provided username
+	// Add adds a new login attempt record for provided username or email
 	Add(ctx context.Context, username, IPAddress string) error
-	// Validate checks if username has to many login attempts inside a window.
+	// ValidateUsername checks if username has to many login attempts inside a window.
 	// Will return true if provided username do not have too many attempts.
-	Validate(ctx context.Context, username string) (bool, error)
+	ValidateUsername(ctx context.Context, username string) (bool, error)
+	// ValidateIP checks if IP address has to many login attempts inside a window.
+	// Will return true if provided IP Address do not have too many attempts.
+	ValidateIPAddress(ctx context.Context, IPAddress string) (bool, error)
```

Comment:
So before we only validated login attempts for username name so ip address validation is something new. Any reason we introduce it for this pr? 

Also should we always validate ip address for all other paths as well? If so we should merge these two into one function

Reply:
Ah, I should've noted that in the PR description. IP validation is introduced as an additional safeguard here to protect against brute force attacks. Since the SMTP client sends an email every time a valid email is entered on the passwordless login form, we wanted a simple way to protect against an attacker iterating through a list of potentially valid email addresses. It's not perfect, since an attacker could spoof an IP address, but seems like a reasonable safeguard. But it probably should be added for all paths, so I'll remove it from this PR and introduce IP validation for all paths in a separate PR.